core/src/main/java/com/linecorp/armeria/common/HttpResponseBuilder.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Locale;
+import java.util.Map.Entry;
+
+import org.reactivestreams.Publisher;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+/**
+ * Builds a new {@link HttpResponse}.
+ */
+public final class HttpResponseBuilder extends AbstractHttpMessageBuilder {
+
+    private final ResponseHeadersBuilder responseHeadersBuilder = ResponseHeaders.builder();
+
+    private final HttpHeadersBuilder httpTrailers = HttpHeaders.builder();
+
```

Comment:
Could you add a package-private constructor to prevent creating directly?
```suggestion
    HttpResponseBuilder() {}
```